parsel/xpathfuncs.py

```diff
@@ -48,6 +49,7 @@ def has_class(context, *classes):
     if node_cls is None:
         return False
     node_cls = ' ' + node_cls + ' '
+    node_cls = re.sub('\s+', ' ', node_cls)
```

Comment:
should we compile this regex?

Reply:
It does seem to bring a small gain though:

```python
>>> %timeit re.sub('\s+', ' ', 'some  random\n text  ')
2.12 µs ± 35.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
>>> p = re.compile('\s+')
>>> %timeit p.sub(' ', 'some  random\n text  ')
1.57 µs ± 13.1 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
What do you think to remove the ugly string concatenation as well?
```
    node_cls = re.sub('\s+', ' ', node_cls).split(' ')
    for cls in classes:
        if cls not in node_cls:
            return False
```

Reply:
I prefer not touching it, as it is outside of the scope of this PR.

And I kind of like the way it is implemented because it exposes in a clear way one of the issues that happen when checking for a CSS class in an element using the "contains" function.

Reply:
If it improves the performance and it is nothing that includes extra complexity in the function, I think there is no harm to compile the regex then. :-)